psutil/arch/windows/socks.c

```diff
@@ -38,13 +40,18 @@ static PVOID __GetExtendedTcpTable(ULONG family) {
     ULONG size;
     TCP_TABLE_CLASS class = TCP_TABLE_OWNER_PID_ALL;
 
+#ifndef Py_GIL_DISABLED
     size = g_TcpTableSize;
     if (size == 0) {
         GetExtendedTcpTable(NULL, &size, FALSE, family, class, 0);
         // reserve 25% more space
         size = size + (size / 2 / 2);
         g_TcpTableSize = size;
     }
+#else
+    GetExtendedTcpTable(NULL, &size, FALSE, family, class, 0);
+    size = size + (size / 2 / 2);
+#endif
```

Comment:
I don't like the fact that here you use 2 different code paths depending on Py_GIL_DISABLED. There should be only one logic.